### PR TITLE
Add support for IPv6 docker host connections.

### LINKED ISSUE #1135: Support IPv6 addresses in DOCKER_HOST
Raised in https://github.com/docker/compose/issues/2879.

See https://github.com/docker/docker/pull/16950 for the Engine implementation.

### PR DESCRIPTION
**Pull request by shin-**
Fixes #1135

### CODE REVIEW

File: tests/unit/utils_test.py [hunk 1]
```diff
@@ -404,18 +404,26 @@ def test_parse_host(self):
...
+            ),
             'somehost.net:80/service/swarm': (
                 'http://somehost.net:80/service/swarm'
             ),
             'npipe:////./pipe/docker_engine': 'npipe:////./pipe/docker_engine',
+            '[fd12::82d1]:2375': 'http://[fd12::82d1]:2375',
+            'https://[fd12:5672::12aa]:1090': 'https://[fd12:5672::12aa]:1090',
+            '[fd12::82d1]:2375/docker/engine': (
+                'http://[fd12::82d1]:2375/docker/engine'
+            ),
```
Comment (dnephin) [inline tests/unit/utils_test.py]: Are these normal ipv6 addresses? I usually see something like `fe80::4216:7eff:fee8:f206/64`

Comment (shin-) [inline tests/unit/utils_test.py]: What do you mean by normal? Both are valid:
http://v6decode.com/#address=fd12%3A%3A82d1
http://v6decode.com/#address=fd12%3A5672%3A%3A12aa

The brackets are conform to [RFC 2732](https://www.ietf.org/rfc/rfc2732.txt)

Comment (dnephin) [inline tests/unit/utils_test.py]: Ah cool, that's what I was wondering. Thanks!

**Comment by dnephin** [comment]
LGTM

**Pull request merged**